File: source/nestjs-petclinic-rest/src/repository/owner.repository.ts
import { Repository, EntityRepository } from 'typeorm';
import { CustomRepository } from './../typeorm/typeorm-ex.decorator';
import { Owner } from './../domain/entities/owner.entity';
import { IntentMerge } from './../intent/intent.decorators';

@IntentMerge()
@CustomRepository(Owner)
export class OwnerRepository extends Repository<Owner> {
}

File: source/spring-petclinic-rest/src/main/java/com/spring_petclinic/spring_petclinic_rest/data/OwnerRepository.java
package com.spring_petclinic.spring_petclinic_rest.data;

import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Collection;
import com.spring_petclinic.spring_petclinic_rest.domain.models.Owner;
import com.spring_petclinic.spring_petclinic_rest.intent.IntentMerge;
import java.util.UUID;

/**
 * Spring Data JPA repository for the Owner entity.
 */
@IntentMerge
public interface OwnerRepository extends JpaRepository<Owner, Integer>{
    Collection<Owner> findByLastName(String lastName);
}